### ISSUE TITLE
[4.3.4] Fix waterwalking bug

### ISSUE THREAD

**Issue by IvanVojtko** [TrinityCore/TrinityCore#10950]
When you die, you can walk on water as a ghost. But when you come to your body and revive yourselve, you can still walk on water. This patch should fix it.
```cpp
--- a/src/server/game/Entities/Player/Player.cpp
+++ b/src/server/game/Entities/Player/Player.cpp
@@ -5179,7 +5179,7 @@ void Player::ResurrectPlayer(float restore_percent, bool applySickness)
 
     setDeathState(ALIVE);
 
-    SetWaterWalking(false);
+    SetWaterWalking(false, true);
     if (!HasUnitState(UNIT_STATE_STUNNED))
         SetRooted(false);
```

**Comment by danlapps** [comment]
Is this still an issue and if so is this a valid patch ready for push ?

**Comment by DDuarte** [comment]
Confirmed that the fix works on a0acd9b11bf6
